sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp

```diff
@@ -0,0 +1,328 @@
+      NullableInt64 = static_cast<int64_t>(4);
+      NullableUint8 = static_cast<uint8_t>(5);
+      NullableUint16 = static_cast<uint16_t>(6);
+      NullableUint32 = static_cast<uint32_t>(7);
+      NullableUint64 = static_cast<uint64_t>(8);
+      NullableFloat = static_cast<float>(9.0);
+      NullableDouble = static_cast<double>(10.0);
+      NullableString = "string";
+
+      Booleans = std::vector<bool>(vectorSize, true);
+      Int8s = std::vector<int8_t>(vectorSize, static_cast<int8_t>(1));
+      Int16s = std::vector<int16_t>(vectorSize, static_cast<int16_t>(2));
+      Int32s = std::vector<int32_t>(vectorSize, static_cast<int32_t>(3));
+      Int64s = std::vector<int64_t>(vectorSize, static_cast<int64_t>(4));
+      Uint8s = std::vector<uint8_t>(vectorSize, static_cast<uint8_t>(5));
+      Uint16s = std::vector<uint16_t>(vectorSize, static_cast<uint16_t>(6));
+      Uint32s = std::vector<uint32_t>(vectorSize, static_cast<uint32_t>(7));
+      Uint64s = std::vector<uint64_t>(vectorSize, static_cast<uint64_t>(8));
+      Floats = std::vector<float>(vectorSize, static_cast<float>(9.0));
+      Doubles = std::vector<double>(vectorSize, static_cast<double>(10.0));
+      Strings = std::vector<std::string>(vectorSize, "string");
+
+      Map = std::map<std::string, std::string>();
+      for (size_t i = 0; i < vectorSize; i++)
+      {
+        Map["key" + std::to_string(i)] = "value" + std::to_string(i);
+      };
+    }
+  };
+
+  /**
+   * @brief Measure the HTTP transport performance.
+   */
+  class JsonTest : public Azure::Perf::PerfTest {
+    enum class Action
+    {
+      Serialize,
+      Deserialize
+    };
+
+    Action m_action;
+    size_t m_vectorSize;
+    JsonTestObject m_testObject;
+    std::string m_jsonBody;
+
+  public:
+    /**
+     * @brief Construct a new JsonTest test.
+     *
+     * @param options The test options.
+     */
+    JsonTest(Azure::Perf::TestOptions options) : PerfTest(options) {}
+
+    void Setup() override
+    {
+      m_action = m_options.GetOptionOrDefault<std::string>("Action", "serialize") == "serialize"
+          ? Action::Serialize
+          : Action::Deserialize;
+
+      m_vectorSize = m_options.GetOptionOrDefault<size_t>("Size", 1000);
+      m_testObject = JsonTestObject(m_vectorSize);
+
+      if (m_action == Action::Deserialize)
+      {
+        m_jsonBody = m_testObject.Serialize();
+      }
+    }
+
+    /**
+     * @brief Perform Json test.
+     *
+     */
+    void Run(Azure::Core::Context const&) override
+    {
+      try
+      {
+        switch (m_action)
+        {
+          case Action::Serialize: {
+            m_testObject.Serialize();
+            break;
+          }
+          case Action::Deserialize: {
+            m_testObject.Deserialize(m_jsonBody);
+            break;
+          }
+        }
+      }
+      catch (std::exception const&)
+      {
+        // don't print exceptions, they are happening at each request, this is the point of the test
```

Comment:
I'm not sure this is valid for this test. Are you really ok with the test swallowing exceptions? And if they are thrown, what does throwing the exception do to the overall measurement? Doesn't throwing an exception introduce a large amount of noise into the results depending on whether it is thrown or not?

Reply:
sure, makes sense

Reply:
Btw, in other cases, I'm a fan of catching exceptions and doing something, but for a perf test, we need to make sure that exceptions aren't thrown (unless we're testing the performance overhead of exceptions ::).